pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -156,6 +156,7 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 			c.Logger.Errorf("Failed to update TaskRun status for PipelineRun %s: %v", pr.Name, err)
 			return err
 		}
+		_ = c.metrics.DurationAndCount(pr)
```

Comment:
Since `DurationAndCount` might take a long time (see comment below), have you considered running this in a goroutine so that `Reconcile` is not blocked?